### ISSUE TITLE
Defining features implements modules.

### ISSUE THREAD

**Issue by kwatsen** [CESNET/libyang#1844]
Given foo.yang:
```
module foo {
  yang-version 1.1;
  namespace "https://example.net/foo";
  prefix "f";

  // Protocol-accessible nodes (ONLY IF MODULE IMPLEMENTED)
  container foos {
    list foo {
      key name;
      min-elements 1;
      leaf name {
        type string;
      }
    }
  }
}
```

The following JSON validates:
```
{
  "foo:foos": {
    "foo": [
      {
        "name": "foo-1"
      }
    ]
  }
}
```

Here's the validation command:
```
# yanglint foo.yang foo.json
# echo $?
0
```

Given bar.yang (same except that it unnecessarily imports `foo`):
```
module bar {
  yang-version 1.1;
  namespace "https://example.net/bar";
  prefix "b";

  import foo {
    prefix "f";
  }

  // Protocol-accessible nodes (ONLY IF MODULE IMPLEMENTED)
  container bars {
    list bar {
      key name;
      min-elements 1;
      leaf name {
        type string;
      }
    }
  }
}
```

The following JSON validates:
```
{
  "bar:bars": {
    "bar": [
      {
        "name": "bar-1"
      }
    ]
  }
}
```

Here's the validation command (note that no `foo` instances are required):
```
# yanglint bar.yang bar.json
# echo $?
0
```

But when the validation command is changed slightly (to add `--features=foo:`), suddenly `foo` instances are required, as if the `foo` module is now implemented, which shouldn't be the case, right?:
```
# yanglint --features=foo: bar.yang bar.json
libyang err : Too few "foo" instances. (Schema location /foo:foos/foo, line number 9.)
YANGLINT[E]: Failed to parse input data file "bar.json".
```

**Comment by michalvasko** [comment]
> as if the foo module is now implemented, which shouldn't be the case, right?

No, it works exactly as intended. It is true that it is not specified explicitly (unfortunately) but we came to the conclusion that you can enable feature only in implemented modules. I do not remember the reasons exactly but the [`grouping name`](https://www.rfc-editor.org/rfc/rfc8525#page-11) in *ietf-yang-library* seems like an adequate justification.

**Comment by kwatsen** [comment]
I don't see what you're pointing on page-11 in RFC 8525.   I don't believe there is anything that states a defined-feature implies that the module is implemented (that would be like the "tail wagging the dog").   

The prior version of YANG Library (RFC 7895) was clearer, as it had separate/distinct `conformance-type` and `feature` list for each module (i.e., these are orthogonal concerns).  The new YANG Library may have a bug in it, as I don't see how to define a feature for an `import-only-module`  (maybe this was your point above?).

Are you trying to say that there is no way to define a `feature` in "import-only" modules?  Now all "import-only" modules must be implemented in order to use features defined in them? 

There crypto-types and friends modules going though the NETCONF WG (past Last Call, past YANG Doctor review, etc.) define groupings with `feature` statements (intended to be used by importing modules) *and* specific-comments stating that the protocol-accessible nodes are only present if the module is implemented...  [PS: all that work was validated using `yanglint 1.x`]

Heck, there is even a feature to indicate when the module is implemented (i.e., `feature central-keystore-supported`):
```
     feature central-keystore-supported {
       description
         "The 'central-keystore-supported' feature indicates that
          the server supports the keystore (i.e., implements the
          'ietf-keystore' module).";
     }

     feature local-definitions-supported {
       description
         "The 'local-definitions-supported' feature indicates that
          the server supports locally-defined keys.";
     }

     feature asymmetric-keys {
       description
         "The 'asymmetric-keys' feature indicates that the server
          supports asymmetric keys in keystores.";
     }

     feature symmetric-keys {
       description
         "The 'symmetric-keys' feature indicates that the server
          supports symmetric keys in keystores.";
     }
```

**Comment by kwatsen** [comment]
I sent an email to the NETMOD mailing list: [here](https://mailarchive.ietf.org/arch/msg/netmod/_w0V47x8fmywvup_EGY0nlTRHII/).

In the meanwhile, do you think you could update the "-VV" output to output the equivalent of its YANG-library?   The current "-VV" output is about how modules are searched for, but there's no "summary" of the net-result (i.e., the context) that `yanglint` is going to use (e.g., for validation)...  Does it seam worthy of -VV output?

**Comment by michalvasko** [comment]
There are so many features now of the non-interactive `yanglint` that one can easily miss them. Look at the help and the description of the `--list` option.

**Comment by kwatsen** [comment]
It looks like the NETMOD WG is mostly with agreement with you.   Lada seems to be holding onto the past but, to be safe, I'm updating all my modules to have a `feature` statement to enable protocol-accessible nodes, which should be supported by `yanglint`.   I've run into an issue, but will open a new GitHub Issue for it.
